### PR TITLE
Trigger on Git merge

### PR DESCRIPTION
**Pull request by nghi-ly**
## What are you changing in this pull request and why?

Docs for beta launch of Trigger on merge:

- new **Merge jobs** page
- update [Deployments landing](https://docs.getdbt.com/docs/deploy/deployments) page
- update [Jobs landing](https://docs.getdbt.com/docs/deploy/jobs) page
- new section **Treatment of merge jobs** on [Job scheduler](https://docs.getdbt.com/docs/deploy/job-scheduler) page
- On [CI jobs](https://docs.getdbt.com/docs/deploy/ci-jobs) page: updated UI labels to match sentence casing; refactored old examples section a bit since the page is getting long 
- new release note

## Checklist
- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [x] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [x] Needs PM review

Adding or removing sections:
- [x] Update link URLs if needed -- checked, not needed, not referenced elsewhere
- [x] Add an entry for deleted sections in `website/vercel.json` if possible -- checked, not needed, not referenced elsewhere
- [x] Run link testing locally with `npm run build` to update the links that point to deleted pages
- [x] Check if screenshot files need to be deleted -- removed when screenshot is not referenced elsewhere

### CODE REVIEW

**Comment by ayshukla** [comment]
This is actually a brand new job type, separate from a CI job. Could we separate this into it's own page? We will then want to update the table on [this](https://docs-getdbt-com-git-ly-docs-trigger-on-merge-dbt-labs.vercel.app/docs/deploy/jobs) page to reflect that.

File: website/docs/docs/deploy/ci-jobs.md [hunk 1]
```diff
@@ -70,26 +77,9 @@ If you're not using dbt Cloud’s native Git integration with [GitHub](/docs/cl
       - `non_native_pull_request_id` (for example, BitBucket)
    - Provide the `git_sha` or `git_branch` to target the correct commit or branch to run the job against. 
 
-## Example pull requests
```
Comment (nghi-ly) [inline website/docs/docs/deploy/ci-jobs.md:73]: cleaned this section up a little bit, page getting a bit long. moved it up to lines 50-58 to be closer to the "set up" section. i think just one example is sufficient to illustrate the integration. 

lemme know what you think! happy to revert

File: website/docs/docs/deploy/ci-jobs.md [hunk 2]
```diff
@@ -38,17 +39,23 @@ To make CI job creation easier, many options on the **CI job** page are set to d
...
+    - **Target name** &mdash; Define the [target name](/docs/build/custom-target-names). Similar to **Environment Variables**, this option lets you customize the behavior of the project. You can use this option to specify that a CI job is running in a _Staging_ or _CI_ environment by setting the target name and modifying your project code to behave differently, depending on the context. 
+    - **Run timeout** &mdash; Cancel this CI job if the run time exceeds the timeout value. You can use this option to help ensure that a CI check doesn't consume too much of your warehouse resources.
+    - **dbt version** &mdash; By default, it’s set to inherit the [dbt version](/docs/dbt-versions/core) from the environment. dbt Labs strongly recommends that you don't change the default setting. This option to change the version at the job level is useful only when you upgrade a project to the next dbt version; otherwise, mismatched versions between the environment and job can lead to confusing behavior.
```
Comment (nghi-ly) [inline website/docs/docs/deploy/ci-jobs.md:46]: the only change in this diff is updating UI labels to be sentence casing (vs title casing). no other content changes.

File: website/docs/docs/deploy/jobs.md [hunk 3]
```diff
@@ -1,23 +1,24 @@
...
 title: "Jobs in dbt Cloud"
 sidebar_label: "About Jobs"
-description: "Learn about deploy jobs and continuous integration (CI) jobs in dbt Cloud and what their differences are." 
+description: "Learn about the different job types in dbt Cloud and what their differences are." 
 tags: [scheduler]
 pagination_next: "docs/deploy/deploy-jobs"
 ---
 
-In dbt Cloud, there are two types of jobs: 
+These are the available job types in dbt Cloud:
```
Comment (nghi-ly) [inline website/docs/docs/deploy/jobs.md:9]: updated to make it more _evergreen_

File: website/static/img/docs/dbt-cloud/using-dbt-cloud/example-create-merge-job.png
Comment (reubenmc) [inline website/static/img/docs/dbt-cloud/using-dbt-cloud/example-create-merge-job.png:1]: There's been a slight UI change during the beta to add the referral settings right below the execution setting, given that the default is state modified. It would probably be good to replace this with the latest screenshot

File: website/docs/docs/deploy/merge-jobs.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+## Set up job trigger on Git merge {#set-up-merge-jobs}
+
+1. On your deployment environment page, click **Create Job** > **Merge Job**. 
+1. Options in the **Job settings** section:
+    - **Job name** &mdash; Specify the name for the merge job.
+    - **Description** &mdash; Provide a descripion about the job. 
+    - **Environment** &mdash; By default, it’s set to your production environment.
+1. In the **Git trigger** section, the **Run on merge** option is enabled by default. Every time a PR merges (to the main branch) in your Git repo, this job will get triggered to run. 
+1. In the **Execution settings** section, add the dbt commands that you want to invoke when this job runs. By default, it includes the `dbt build --select state:modified+` command. This informs dbt Cloud to build only new or changed models and their downstream dependents. For more [commands](/docs/deploy/job-commands), click **Add command**.
```
Comment (reubenmc) [inline website/docs/docs/deploy/merge-jobs.md]: Probably good to call out here that if using `state:modified+` they will need to set `Compare changes against an environment (Deferral)` which is automatically set to the current enviroment (and this is right below this execution settings now rather than buried in advanced settings which was a mistake we discovered from the beta).

**Review by reubenmc — APPROVED** [review]
Looks great! A couple of small comments on deferral
Screenshot 2024-04-16 at 10 17 12 AM
 settings.

File: website/static/img/docs/dbt-cloud/using-dbt-cloud/example-create-merge-job.png
Comment (nghi-ly) [reply website/static/img/docs/dbt-cloud/using-dbt-cloud/example-create-merge-job.png:1]: thanks for the heads up! will recapture this

File: website/docs/docs/dbt-versions/release-notes/70-Apr-2024/merge-jobs.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+description: "April 2024: Native support now available in dbt Cloud to trigger dbt job runs when Git pull requests merge."
+sidebar_label: "New: Merge jobs"
+sidebar_position: 07
+tags: [Apr-2024]
+date: 2024-04-16
+---
+
+# New: Merge jobs <Lifecycle status="beta" />
+
+You can now set up a continuous deployment (CD) workflow for your projects natively in dbt Cloud. A beta release of [Merge jobs](/docs/deploy/merge-jobs), a new [job type](/docs/deploy/jobs), is available that enables you to trigger dbt job runs as soon as changes (via Git pull requests) merge into production.
```
Comment (runleonarun) [inline website/docs/docs/dbt-versions/release-notes/70-Apr-2024/merge-jobs.md]: Tiny sugg because I read this as a list. 
```suggestion
You can now set up a continuous deployment (CD) workflow for your projects natively in dbt Cloud. You can now access a beta release of [Merge jobs](/docs/deploy/merge-jobs), which is a new [job type](/docs/deploy/jobs), that enables you to trigger dbt job runs as soon as changes (via Git pull requests) merge into production. 
```

File: website/docs/docs/deploy/merge-jobs.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
+---
+title: "Merge jobs in dbt Cloud"
+sidebar_label: "Merge jobs"
+description: "Learn how to trigger a dbt job run when a Git pull request merges."
+---
+
+# Merge jobs in dbt Cloud <Lifecycle status="beta" />
+
+To implement a continuous development (CD) workflow in dbt Cloud, you can set up merge jobs that triggers a dbt job to run when Git pull requests are merged into production. This creates a seamless development experience where changes made in code will automatically update production data.
```
Comment (runleonarun) [inline website/docs/docs/deploy/merge-jobs.md]: Suggest breaking up this sentence a tiny bit:
```suggestion
You can set up a merge job to implement a continuous development (CD) workflow in dbt Cloud. The merge job triggers a dbt job to run when someone merges Git pull requests into production. This creates a seamless development experience where changes made in code will automatically update production data. 
```

File: website/docs/docs/deploy/merge-jobs.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+## Set up job trigger on Git merge {#set-up-merge-jobs}
+
+1. On your deployment environment page, click **Create Job** > **Merge Job**. 
+1. Options in the **Job settings** section:
+    - **Job name** &mdash; Specify the name for the merge job.
+    - **Description** &mdash; Provide a descripion about the job. 
+    - **Environment** &mdash; By default, it’s set to the environment you created the job from.
+1. In the **Git trigger** section, the **Run on merge** option is enabled by default. Every time a PR merges (to the main branch) in your Git repo, this job will get triggered to run. 
+1. In the **Execution settings** section, add the dbt commands that you want to invoke when this job runs. By default, it includes the `dbt build --select state:modified+` command. This informs dbt Cloud to build only new or changed models and their downstream dependents. For more [commands](/docs/deploy/job-commands), click **Add command**.
```
Comment (runleonarun) [inline website/docs/docs/deploy/merge-jobs.md]: @nghi-ly Is this [repeated below](https://github.com/dbt-labs/docs.getdbt.com/pull/5244/files#diff-f28b1583630dc788056299dd4237e262b2417c0fd769d89a272c57e4bb65a08bR27) for a reason?

File: website/docs/docs/deploy/jobs.md [hunk 8]
```diff
@@ -1,23 +1,24 @@
...
-| --- | --- | --- |
-| Purpose | Builds production data assets. | Builds and tests new code before merging changes into production. |
-| Trigger types | Triggered by a schedule or by API. | Triggered by a commit to a PR or by API. |
-| Destination | Builds into a production database and schema. | Builds into a staging database and ephemeral schema, lived for the lifetime of the PR. |
-| Execution mode | Runs execute sequentially, so as to not have collisions on the underlying DAG. | Runs execute in parallel to promote team velocity. |
-| Efficiency run savings | Detects over-scheduled jobs and cancels unnecessary runs to avoid queue clog. | Cancels existing runs when a newer commit is pushed to avoid redundant work. |
-| State comparison | Only sometimes needs to detect state. | Almost always needs to compare state against the production environment to build on modified code and its dependents. |
-| Job run duration | Limit is 24 hours. | Limit is 24 hours. |
\ No newline at end of file
+|  | Deploy jobs | CI jobs | Merge jobs |
```
Comment (runleonarun) [inline website/docs/docs/deploy/jobs.md]: I don't have any great suggestions but this table is a little hard to parse at first because the job types are along the top. I see why we've done that (it would look silly to have 7 columns and 3 rows.)

Maybe just bolding the first column would help?

File: website/docs/docs/deploy/job-scheduler.md [hunk 9]
```diff
@@ -57,6 +57,12 @@ When compared to deployment jobs, the scheduler behaves differently when handlin
 - **Will the CI run consume a run slot?** &mdash; CI runs don't consume run slots and will never block production runs.
 - **Does this same job have a run already in progress?** &mdash; CI runs can execute concurrently (in parallel). CI runs build into unique temporary schemas, and CI checks execute in parallel to help increase team productivity. Teammates never have to wait to get a CI check review.
 
+### Treatment of merge jobs
+The scheduler queues a [merge job](/docs/deploy/merge-jobs) to be processed when it's triggered to run by a _merged_ Git pull request.
```
Comment (runleonarun) [inline website/docs/docs/deploy/job-scheduler.md]: Suggesting writing in the order it happens to make it easier to understand.
```suggestion
When triggered by a _merged_ Git pull request, the scheduler queues a [merge job](/docs/deploy/merge-jobs) to be processed.
```

**Review by runleonarun — APPROVED** [review]
Just some small suggestions! Overall looks great!

File: website/docs/docs/deploy/merge-jobs.md (hunk 7 above)
Comment (nghi-ly) [reply website/docs/docs/deploy/merge-jobs.md]: great catch! thank you!

File: website/docs/docs/deploy/merge-jobs.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
```
Comment (nghi-ly) [inline website/docs/docs/deploy/merge-jobs.md]:
```suggestion
```

File: website/docs/docs/deploy/jobs.md (hunk 8 above)
Comment (nghi-ly) [inline website/docs/docs/deploy/jobs.md]:
```suggestion
|  | **Deploy jobs** | **CI jobs** | **Merge jobs** |  
```

File: website/docs/docs/deploy/merge-jobs.md (hunk 7 above)
Comment (nghi-ly) [reply website/docs/docs/deploy/merge-jobs.md]: removed duplicate

File: website/docs/docs/dbt-versions/release-notes/70-Apr-2024/merge-jobs.md [hunk 11]
```diff
@@ -0,0 +1,14 @@
+---
+title: "New: Merge jobs"
+description: "April 2024: Native support now available in dbt Cloud to trigger dbt job runs when Git pull requests merge."
+sidebar_label: "New: Merge jobs"
+sidebar_position: 07
+tags: [Apr-2024]
+date: 2024-04-16
```
Comment (nghi-ly) [inline website/docs/docs/dbt-versions/release-notes/70-Apr-2024/merge-jobs.md]:
```suggestion
date: 2024-04-17
```
